### PR TITLE
SDN-4114: Do per-pod MCS/metadata blocking with nftables rather than iptables

### PR DESCRIPTION
**Pull request by danwinship**
Along with warning about customers using iptables in pods (https://issues.redhat.com/browse/SDN-4114) we need to make sure that we _don't_ use iptables in pods.

This only changes our per-pod iptables rules to use nftables; openshift-sdn still uses iptables in the host network namespace for now.

### CODE REVIEW

**Comment by danwinship** [comment]
Trying to filter out "expected" iptables rules from the iptables alerter turns out to not work because of iptables version incompatibilities, so we need this

**Comment by npinaeva** [comment]
the code looks correct, but I don't trust my manual nftables rules parsing that much, so are these rules tested somehow in CI (or were they tested manually)?

**Comment by danwinship** [comment]
Rules were tested manually. It looks like we don't have any e2e tests for it (since this was always supposed to have been a temporary hack that was going away soon) but I know QE tests it.

File: pkg/cmd/openshift-sdn-cni/openshift-sdn.go [hunk 1]
```diff
@@ -116,24 +116,73 @@ func (p *cniPlugin) testCmdAdd(args *skel.CmdArgs) (types.Result, error) {
 	return convertToRequestedVersion(args.StdinData, result)
 }
 
-func generateIPTablesCommands(platformType string) [][]string {
+func doNFTablesRules(platformType string) error {
+	nft, err := knftables.New(knftables.IPv4Family, "openshift-block-output")
```
[the remark this answers is not in the document]
Comment (danwinship) [reply pkg/cmd/openshift-sdn-cni/openshift-sdn.go:120]: It's in the pod network namespace, so "pod" is implied I think

[the remark this answers is not in the document]
Comment (npinaeva) [reply pkg/cmd/openshift-sdn-cni/openshift-sdn.go:120]: right, I think it is more obvious when looking on the real rules than from the code, fine with me either way

**Pull request merged**